### ISSUE TITLE
Schema inference and Nullble types in Parquet

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#35346]
(version 22.2.2)

```sql
cd /var/lib/clickhouse/user_files/

------- String 
clickhouse-client -q "select Null::Nullable(String) a format Parquet" > nullable.Parquet

clickhouse-client -q "select * from file('nullable.Parquet', 'Parquet', ' a Nullable(String)')"
\N

clickhouse-client -q "select * from file('nullable.Parquet', 'Parquet')";
    DB::Exception: Cannot convert NULL value to non-Nullable type: while converting column `a` from type Nullable(String) to type String

------- Float64 
clickhouse-client -q "select Null::Nullable(Float64) a format Parquet" > nullable.Parquet

clickhouse-client -q "select * from file('nullable.Parquet', 'Parquet')"
   DB::Exception: Cannot convert NULL value to non-Nullable type: while converting column `a` from type Nullable(Float64) to type Float64
```

I guess the same with any type.

**Comment by k138z** [comment]
Hello, I would like to solve this issue since it is blocking some of our applications, but I have no idea where to start. Any pointers/references would help a lot. Thanks

**Comment by dchimeno** [comment]
I also faced this issue in version: 22.3.2.1
A workaround is to put the structure (schema) in the query as mentioned above.

**Comment by den-crane** [comment]
it was fixed in 22.4 https://github.com/ClickHouse/ClickHouse/pull/35582
